### DISCUSSION TITLE
[datafeeder]: Edit wizzard form configuration

### DISCUSSION THREAD

**Discussion by fgravin** [geonetwork/geonetwork-ui#70] [Ideas]
Datafeeder will propose a very simple form to fill the metadata record as designed in the following [mockups](https://xd.adobe.com/view/1204e7a0-ff18-498b-a67e-87ba0b953947-4757/screen/01c82a4e-5560-4e6e-b90f-3634a025bbce?fullscreen).

I think we need to configure this form instead of having static hard coded component. Here a model we can discuss about, it has to be simple.

```js
const formConfig = [
  {
    // step 1
    fields: [
      {
        id: 'title',
        label: 'datafeeder.form.title',
        icon: '....',
        type: 'text',
      },
      {
        id: 'abstract',
        label: 'datafeeder.form.abstract',
        icon: '....',
        type: 'textarea',
      },
    ],
  },
  {
    // step 2
    fields: [
      {
        id: 'tags',
        label: 'datafeeder.form.tags',
        icon: '....',
        type: 'keyword-picker',
      },
    ],
  },
]
```

At some point, it will probably need something to define how it will be integrated in the metadata record (eg. xpath).

Please share your opinion about that.
Thanks

**Comment by fxprunayre** [comment]
Sounds good to have a config.
* why not adding the step config (label?)
* how do you plan form validation, errors ? 
* probably not perfect but some terms or ideas to reuse from https://github.com/geonetwork/core-geonetwork/blob/master/schemas/config-editor.xsd (tab can be considered as step, help text)


> At some point, it will probably need something to define how it will be integrated in the metadata record (eg. xpath).

Not sure how you plan to build XML from this, but maybe it make more sense to have an XML template which reference the form fields ?
```xml
<cit:CI_Citation>
  <cit:title>
    <gco:CharacterString>{{datafeeder.form.title}}</gco:CharacterString>
```

 Also I suppose the analysis will extract data model for the source dataset, so you should consider using ISO19115-3 which embeds feature catalogue in the same metadata record (but that is not related to the wizard form).

**Comment by pvgenuchten** [comment]
Hi @fgravin, i like this idea of basic dataset registration

- Starting with an existing dataset (uploaded) or service makes a lot of sense, because you can already take some of the embedded metadata as a start.
- Having a fixed metadata-model helps to make form validation much more simple

We did a [similar implementation](https://github.com/GeoCat/core-geonetwork/tree/geodatastore) some years ago


![image](https://user-images.githubusercontent.com/299829/102374723-f49b7980-3fc1-11eb-8b0e-fbc9dc3fc2f6.png)

A challenge to integrate this into/next to geonetwork is when people want to modify records created with datafeeder in an advanced editor (or bulk edit) and the record is later reopened in the data feeder ui, do you give some warning that recent changes will be discarded?

To me this is a fully valid approach of metadata creation.
